### PR TITLE
sql: Re-prompt user for input after unrecognized command

### PR DESCRIPTION
**Pull request by NickCellino**
This fixes a bug in the SQL REPL where after a user enters an
unrecognized command, the REPL would not print another "> " prompt and
would not accept any more input.

### CODE REVIEW

File: Userland/Utilities/sql.cpp [hunk 1]
```diff
@@ -304,22 +304,21 @@ class SQLRepl {
...
+        } else {
+            if (command.starts_with(".read ")) {
+                if (!m_input_file) {
+                    auto parts = command.split_view(' ');
+                    if (parts.size() == 2)
+                        source_file(parts[1]);
+                    else
+                        outln("\033[33;1mUsage: .read <sql file>\033[0m");
+                } else
+                    outln("\033[33;1mCannot recursively read sql files\033[0m");
```
Comment (alimpfard) [inline Userland/Utilities/sql.cpp:320]: Please put braces around both branches, or neither if both are single-line.

File: Userland/Utilities/sql.cpp [hunk 2]
```diff
@@ -304,22 +304,21 @@ class SQLRepl {
...
+            } else
+                outln("\033[33;1mUnrecognized command:\033[0m {}", command);
```
Comment (alimpfard) [inline Userland/Utilities/sql.cpp:322]: Same here.

File: Userland/Utilities/sql.cpp [hunk 3]
```diff
@@ -302,8 +302,12 @@ class SQLRepl {
             auto parts = command.split_view(' ');
             if (parts.size() == 2)
                 connect(parts[1]);
-            else
+            else {
                 outln("\033[33;1mUsage: .connect <database name>\033[0m");
```
Comment (alimpfard) [inline Userland/Utilities/sql.cpp:306]: And here.

**Comment by trflynn89** [comment]
Seems like it'd be a bit nicer to just make `handle_command` return whether it was actually able to handle the command, and if it returns false, do the `deferred_invoke` directly from the call site in `read_sql` instead?

File: Userland/Utilities/sql.cpp (hunk 1 above)
Comment (NickCellino) [reply Userland/Utilities/sql.cpp:320]: roger that

**Comment by NickCellino** [comment]
> Seems like it'd be a bit nicer to just make `handle_command` return whether it was actually able to handle the command, and if it returns false, do the `deferred_invoke` directly from the call site in `read_sql` instead?

Hello @trflynn89. I basically took your advice here, but instead of "was it able to handle the command or not", I was thinking of it more as "are we ready for more input immediately or are we waiting on some asynchronous action". If there's a better way to do it, let me know :)

**Comment by trflynn89** [comment]
Sure, that sounds fine :)

**Pull request merged**